bacon/model/Tags/TagManager.swift

```diff
@@ -564,13 +563,13 @@ extension TagManager {
     /// Removes a Tag. This method automatically updates data stores and saves to disk.
     /// If a parent Tag is removed, all of its children Tags will be removed too.
     /// - Requires: The Tag being removed must exist.
-    private func removeTag(_ displayValue: String, of parentDisplayValue: String? = nil) {
+    @discardableResult
+    private func removeTag(_ displayValue: String,
```

Comment:
@GilgameshTC Previously, I mentioned that I wanted the API method to do the necessary guarding before calling the private util methods. I realized one more benefit of that is that it ensures that the operation will be atomic, either it completes successfully or doesn't start with.

Do you think that modifying it will break this atomicity? Like will we end up in a limbo state, e.g. parent removed but child not removed?